### ISSUE TITLE
Slight change in love.audio.play() and :play()

### ISSUE THREAD

**Issue by slime73** [love2d/love#309]
**[Original report](https://bitbucket.org/rude/love/issue/304) by Anonymous.**

----------------------------------------

There are numerous topic on the forums of people not understanding how the LOVE audio module works. Most of these issues stem from calling play multiple times on an audio source and expecting to hear multiple sounds. If you call play alone multiple times on a source it will not actually play again until the source has stopped, and I understand this is the current expected behavior.

I find that if you call :stop() before calling :play() on any source then you can get rapidly repeating sounds without having to create multiple sources or rely on an audio lib (Like TEsound) to do so for you.

Basically, I propose that love.audio.play() and :play() call a stop method  on the source before playing it. I have attached a love file demonstrating what I mean. I believe this change would eliminate most if not all of the confusion by our new LOVErs.
